### PR TITLE
use new Arcade publishing infrastructure

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

**Comment by jonsequitur** [comment]
@JohnTortugo

File: .vsts-ci.yml [hunk 1]
```diff
@@ -15,134 +15,141 @@ pr:
...
-              _BuildConfig: Debug
-              _SignType: test
-              _DotNetPublishToBlobFeed: false
-              _BuildArgs:
+stages:
+- stage: build
+  displayName: Build and Test
+  jobs:
+  - template: /eng/common/templates/jobs/jobs.yml
+    parameters:
```
Comment (JohnTortugo) [inline .vsts-ci.yml]: You need to pass `enablePublishUsingPipelines: true` here. See reference here: https://github.com/dotnet/arcade/pull/4556/files?short_path=5153417#diff-c88223c5b65116dd5625673d42760adfR77

File: .vsts-ci.yml [hunk 2]
```diff
@@ -15,134 +15,141 @@ pr:
...
+        variables:
+        # Only enable publishing in official builds.
+        - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+          # DotNet-Blob-Feed provides: dotnetfeed-storage-access-key-1
+          # Publish-Build-Assets provides: MaestroAccessToken, BotAccount-dotnet-maestro-bot-PAT
+          - group: DotNet-Blob-Feed
+          - group: DotNet-Symbol-Server-Pats
+          - group: Publish-Build-Assets
+          - name: _OfficialBuildArgs
+            value: /p:DotNetSignType=$(_SignType)
```
Comment (JohnTortugo) [inline .vsts-ci.yml]: For now you'll need to add pass these two properties as well here: `DotNetPublishUsingPipelines` and `DotNetArtifactsCategory`. See example here: https://github.com/dotnet/arcade/blob/master/eng/common-variables.yml#L37

File: .vsts-ci.yml [hunk 3]
```diff
@@ -15,134 +15,141 @@ pr:
...
+                  /p:TeamName=$(_TeamName)
+                  /p:DotNetPublishBlobFeedKey=$(dotnetfeed-storage-access-key-1)
+                  /p:DotNetPublishBlobFeedUrl=https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
+                  /p:DotNetPublishToBlobFeed=$(_DotNetPublishToBlobFeed)
+                  /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+                  /p:PublishToSymbolServer=true
+                  /p:DotNetSymbolServerTokenMsdl=$(microsoft-symbol-server-pat)
+                  /p:DotNetSymbolServerTokenSymWeb=$(symweb-symbol-server-pat)
+          - name: _DotNetArtifactsCategory
```
Comment (JohnTortugo) [inline .vsts-ci.yml]: Seems like this is unused right now in _your_ build-def. I think you should move it to the outer most scope of the YAML and use them whenever you need to pass their value as parameters to other YAML and/or in `OfficialBuildArgs`.

File: .vsts-ci.yml [hunk 4]
```diff
@@ -15,134 +15,141 @@ pr:
...
-          --configuration $(_BuildConfig)
-          --prepareMachine
-        displayName: Build
+        steps:
+        - checkout: self
+          clean: true
+        - script: eng/common/cibuild.sh
+            --configuration $(_BuildConfig)
+            --prepareMachine
+          displayName: Build
```
Comment (JohnTortugo) [inline .vsts-ci.yml]: You need to include the `post-build.yml` template at the end of the build definition. See example here: https://github.com/dotnet/arcade/pull/4556/files#diff-c88223c5b65116dd5625673d42760adfR140

**Comment by JohnTortugo** [comment]
Added a few comments. Also note that you'll need to make sure that your builds have a default channel in Maestro++. See guideline here: https://github.com/dotnet/arcade/pull/4556/files#diff-c88223c5b65116dd5625673d42760adfR298

File: .vsts-ci.yml (hunk 3 above)
Comment (JohnTortugo) [reply .vsts-ci.yml]: This variable _needs_ to be at the top level of your build-def.

File: .vsts-ci.yml [hunk 5]
```diff
@@ -1,6 +1,8 @@
 variables:
   - name: _TeamName
-    value: DotNetCore
+  value: DotNetCore
```
Comment (JohnTortugo) [inline .vsts-ci.yml]: syntax error here. Indentation.

**Pull request merged**